spring-session/src/main/java/org/springframework/session/data/mongo/MongoExpiringSession.java

```diff
@@ -0,0 +1,125 @@
+package org.springframework.session.data.mongo;
+
+import org.springframework.session.ExpiringSession;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Session object providing additional information about
+ * the datetime of expiration
+ *
+ * @author Jakub Kubrynski
+ * @since 1.2
+ */
+public class MongoExpiringSession implements ExpiringSession {
```

Comment:
Not if we want to allow users to create their own converters -> then then need to operate on this object